### ISSUE TITLE
Verbrenner-Vergleich

### ISSUE THREAD

**Issue by ledermann** [solectrus/solectrus#2407]
Diskutiert in #2402:

Wer ein Elektroauto hat und an seiner PV-Anlage lädt, ist vermutlich auch daran interessiert, einen Kosten-Vergleich zu einem (fiktiven) Verbrennerauto zu ziehen.

Eine einfache Berechnung könnte so erfolgen:

"Du hast im gewählten Zeitraum 134 kWh ins E-Auto geladen. Damit konntest du 1.030 km weit fahren (wegen 13 kWh/100km). Ein Verbrenner hätte für die gleiche Strecke 72 L Diesel verbraucht (wegen 7L/100km) und dich 130 € gekostet (wegen 1,80€/L)."

Grafisch könnte das als simples Balkendiagramm dargestellt werden: Ein Balken zeigt das E-Auto, ein weiterer den Verbrenner. Die jeweiligen Kosten und der Unterschied würden in EUR dargestellt.

Benötigt werden dafür einige zusätzliche Angaben:

- Verbrauch des genutzten E-Autos in kWh/100 km
- Verbrauch eines fiktiven Verbrenners in L/100 km
- Spritpreisentwicklung im Zeitverlauf, ggfs. Ermittlung über eine API (z.B. https://creativecommons.tankerkoenig.de/), notfalls könnte man aber auch mit einem konstanten Wert rechnen.

Die Berechnung könnte noch verfeinert werden: Würde man einen Verbrenner fahren, würde man weniger Strom verbrauchen und somit weniger Strom einkaufen und mehr einspeisen. Beide Beträge (Reduzierung des Stromzukaufs sowie Erhöhung der Einspeisevergütung) könnten berechnet werden und würden die Verbrenner-Vergleichskosten reduzieren.

Damit die Berechnung erfolgen kann, müssen natürlich die einzelnen Ladevorgänge der Wallbox mit ihren Strommengen bekannt sein. Bei Verwendung des SENEC-Speichers in Verbindung mit der Original-SENEC-Wallbox liegen die Werte vor. Bei einer anderen Wallbox oder einem anderen Speicher müssen die Werte anderweitig ermittelt werden, das wäre aber ein separates Thema.

**Comment by halmich** [comment]
Meine Idee für eine "schnelle" und einfache Umsetzung für Anwender, da nicht für jeden die entsprechenden Werte aus dem E-Auto abrufbar sind.

Der Anwender gibt folgende Daten in Solectrus ein (entweder in den Settings ähnlich dem Strompreis, oder über die .env):
- Verbrauch des E-Autos (z.B. 16 kWh / 100km), evtl. pro Quartal, da im Winter der Verbrauch höher ist.
- Verbrauch des (fiktiven) Verbrenners: z.B. 5 l / 100 km
- Preis für 1 Liter Diesel / Benzin (entweder über eine API (s.o.), oder vom User als Eingabe (z.B. 1,60 Euro)

Solectrus berechnet (im Hintergrund), wie weit man mit einer 1 kWh Strom elektrisch fahren kann (100 km / 16 = 6,25km) und was man dafür mit einem Verbrenner bezahlen müsste (5 x 1,60 / 100 * 6,25 = 0,50 Euro).
Man sieht die Ersparnis, 1 kWh Strom aus PV Energie spart 50 Cent Kosten an einer echten Tankstelle. 
Man sieht sogar: 1 kWh Strom aus dem Netz für z.B. 30 Cent spart ebenfalls Geld (bei den obigen Werten).

**Comment by ledermann** [comment]
@halmich Ja, so wäre das machbar. Entschieden ist hier noch nichts, ich sammele noch Ideen. Gut finde ich den Vorschlag, auch bei fehlenden Auto-Messwerten rechnen zu können (durch manuelle Eingabe von Verbrauch und Fahrleistung).

Ein Detail nur: Anstelle von "wie weit kommt man mit 1 kWh Strom" würde ich eher vergleichen wollen, was 1 km Fahrt kostet. Das erscheint mir leichter verständlich zu sein.

Insgesamt sehe ich momentan drei Szenarien, die verglichen werden könnten:

1) Aktuelle, tatsächliche Situation: PV-Anlage mit E-Auto (PowerSplitter berechnet Anteil des grünen Stroms, siehe #2436)
2) Fiktive Situation: PV-Anlage mit Verbrenner (unter der Annahme einer identischen Fahrleistung und einem angenommenen Spritverbrauch)
3) Fiktive Situation: Keine PV-Anlage, aber E-Auto (wie 1, aber mit 100 % Netzbezug)

Für jedes Szenario würden die Kosten pro km berechnet werden und wären sehr leicht vergleichbar.
